File: G_Palindrome_Array.py
N = int(input())

# A = []
# for x in input().split():
#     A.append(int(x))

# A = [int(x) for x in input().split()]
A = list(map(int, input().split()))

isPalindrome = True

for i in range(N//2):
    if A[i] != A[N-i-1]:
        isPalindrome = False
        break

if isPalindrome:
    print("YES")
else:
    print("NO")

File: G_Palindrome_Array.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
    int n;
    scanf("%d", &n);
    int a[n], i;
    for (i = 0; i < n; i++)
        scanf("%d", &a[i]);
    bool isPalindrome = true;
    for (i = 0; i < n / 2; i++)
    {
        if (a[i] != a[n - i - 1])
        {
            isPalindrome = false;
            break;
        }
    }
    if (isPalindrome)
        printf("YES\n");
    else
        printf("NO\n");
    return 0;
}